Added estimateOverlap

TuomasBorman opened the pull request:
Hi,

here is the implementation of [`microbiome::overlap`](https://github.com/microbiome/microbiome/blob/master/R/overlap.R). 

**I compared `microbiome::overlap` with [`Gao's overlap`](https://github.com/Russel88/DOC/blob/master/R/DOC.do.R)**

- 	Gao returns a list of overlap and **rJSD (root Jensen-Shannon divergence)** (and a dataframe with Overlap and rJSD as vectors) / microbiome only overlap
- 	**Overlap is similar**, microbiome has only option for relabundance, Gao does not restrict this
- 	Gao does not have detection threshold

**Jensen-Shannon divergence**

- 	We have this already implemented --> `calculateJSD`
- 	rJSD in matrix format `sqrt(as.matrix(calculateJSD(tse)))`
- 	However, this method does not have detection threshold, should there be one?
- 	(Man page does not have a formula, I can add)
- 	**Is this commonly used as an overlap estimate? If so, should estimateOverlap have new parameter `index`(?) that have options overlap, JSD, and rJSD? `estimateOverlap(x, index = c("overlap", "JSD", rJSD"), detection = 0)`**

Issue: https://github.com/microbiome/mia/issues/60

-Tuomas

TuomasBorman committed:
Implemented microbiome::overlap

TuomasBorman committed:
Added JSD and rJSD

TuomasBorman committed:
Removed JSD and rJSD

TuomasBorman committed:
Modified example

TuomasBorman committed:
calculateJSD: exprs_values --> abund_values

TuomasBorman committed:
all exprs_values --> abund_values

antagomir commented:
Can we view "overlap" as one type of dissimilarity matrix? If there are no relevant differences, we could implement this in the same way than `calculateJSD`, `calculateUnifrac`..  in that case the name could also be `calculateOverlap`.. ?

FelixErnst reviewed (changes requested):
Hi Tuomas,

thanks for the PR.

Can you seperate the renaming things from the implementation. I don't think there is a link between the two things, is there?

Also remember, that you have to support legacy code now. This renaming could potentially break someones code. Just add an `abund_values = exprs_values` as an argument and use the abund_values argument internally. Modify the man pages accordingly.

Best
Felix

TuomasBorman committed:
Revert exprs_values -> abund_values

TuomasBorman committed:
Return value distances, renaming

TuomasBorman committed:
Added info to man page

TuomasBorman committed:
For+for --> apply

TuomasBorman committed:
Added tests

FelixErnst commented on R/calculateOverlap.R:
```diff
@@ -0,0 +1,109 @@
+  # Overlap is the average of the sums of the values in each sample
+  overlap <- (sum(x) + sum(y))/2
+  return(overlap)
+}
```
Can you fix the indentation, bump the version, amend the NEWS and merge?

TuomasBorman committed:
Sync

TuomasBorman committed:
Indentation fix

TuomasBorman committed:
version bump, updated news

FelixErnst reviewed (changes requested):
Another thing: Shouldn't the function be called `estimateOverlap` as the PR title suggests?

@antagomir

FelixErnst commented:
Any news on this regarding the function name?

antagomir commented:
> Another thing: Shouldn't the function be called `estimateOverlap` as the PR title suggests?

My earlier comment: "Can we view "overlap" as one type of dissimilarity matrix? If there are no relevant differences, we could implement this in the same way than `calculateJSD`, `calculateUnifrac`.. in that case the name could also be `calculateOverlap`.. ?" -> the title suggested `estimateOverlap` but perhaps we can change this now to make it consistent with the other dissimilarity measures?

TuomasBorman committed:
Sync

Sync

# Conflicts:
#	DESCRIPTION
#	NEWS
#	tests/testthat/test-5calculateDistance.R

FelixErnst commented:
test it, bump it, NEWS add it (Think about that one Daft Punk song, when reading)

TuomasBorman committed:
Sync

Sync

# Conflicts:
#	NEWS

TuomasBorman committed:
Added tests

TuomasBorman commented:
Lol, I had to play it

TuomasBorman committed:
Fix

TuomasBorman committed:
fix

TuomasBorman committed:
Fix

TuomasBorman committed:
Added runOverlap

TuomasBorman commented:
I added `runOverlap`. It stores result to `reducedDim`.

TuomasBorman committed:
Sync

TuomasBorman committed:
Version bump
